Introducing `absoluteStrokeWidth` option on Lucide Components

ericfennis opened the pull request:
closes #979 #976

Introducing `scaleStrokeWidth`, an option to control the behavior of scaling `strokeWidth` in icons. A new option to control the look of Lucide icons.

By default, scaling up SVG icons will cause the `strokeWidth` to scale with the `viewBox` set on the SVG icon.
Setting `scaleStrokeWidth` to `false`, the `strokeWidth` will remain the same size visually.

## Visualised example
**Enabled** (Default, Current behaviour, SVG default behaviour)
`scaleStrokeWidth: true`

![image](https://user-images.githubusercontent.com/11825403/217630671-264a1447-b23a-44d0-a8ff-4f10932efeb8.png)

**Disabled** (keeping the set strokeWidth in pixels)
`scaleStrokeWidth: false`

![image](https://user-images.githubusercontent.com/11825403/217630577-fac08426-4aba-4d09-a2f7-ae961f07abf7.png)

## React Code Example

```js
import { Camera } from 'lucide-react';

const Compo = () => {
  return <Camera color="red" size={48} scaleStrokeWidth={false} />;
};

export default App;
```

ericfennis commented on packages/lucide-preact/src/createLucideIcon.ts:
```diff
@@ -30,7 +31,7 @@ const createLucideIcon = (iconName: string, iconNode: IconNode): FunctionCompone
         height: size,
         stroke: color,
-        ['stroke-width' as 'strokeWidth']: strokeWidth,
+        ['stroke-width' as 'strokeWidth']: scaleStrokeWidth ? strokeWidth : Number(strokeWidth) / (Number(size) / 24),
```
Hmm, We should not round the numbers, because the `strokeWidth` should have a float.
According to the float precision we could use toFixed() to convert it to maximum decimals.
I can take a look to your alternative calculation.

karsa-mistmere replied:
To fixed will always show decimals, e.g. `4.000`.

ericfennis commented on packages/lucide-preact/scripts/buildTypes.mjs:
```diff
@@ -31,6 +31,7 @@ interface LucideProps extends Partial<Omit<JSX.SVGAttributes, "ref" | "size">> {
   color?: string
   size?: string | number
   strokeWidth?: string | number
+  scaleStrokeWidth?: boolean
```
I was in for an easy name not too difficult to remember.
I'm not really a fan of the`disable[Stroke]Scaling`. 
But `absoluteStrokeWidth` is also a good candidate for this new feature.

ericfennis commented:
@karsa-mistmere I renamed `scaleStrokeWidth` to `absoluteStrokeWidth`.
I think the way of defining is way better now. 

This changed from this:
```jsx
<HomeIcon
  scaleStrokeWidth={false}
  strokeWidth={1}
  size={32}
/>
```

To this
```jsx
<HomeIcon
  absoluteStrokeWidth
  strokeWidth={1}
  size={32}
/>
```

I think this is better.

Also added the calculation you provided!

karsa-mistmere commented:
Cool, that's exactly why I was pushing for opting in instead of opting out, the React code is just a bit cleaner. ^_^ 

I've also updated the Angular package to match, see #949

ericfennis commented:
@karsa-mistmere Awesome work!
